src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,71 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.exception.UserNotAllowed;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private RegistrationService registrationService = new RegistrationServiceImpl();
+    private User user;
+
+    @BeforeEach
+    void setup() {
+        user = new User();
+        user.setLogin("volodymyr.nyshta@gmail.com");
+        user.setAge(21);
+        user.setPassword("123Test");
+    }
+
+    @Test
+    void userWithTheSameLoginAlreadyExists_NotOk() {
+        User user2 = new User();
+        user2.setAge(18);
+        user2.setLogin("volodymyr.nyshta@gmail.com");
+        user2.setPassword("1234567");
+        registrationService.register(user);
+        assertThrows(UserNotAllowed.class, () -> registrationService.register(user2));
+    }
+
+    @Test
+    void userNullNotOk() {
+        assertThrows(UserNotAllowed.class, () -> registrationService.register(null));
+    }
+
+    @Test
+    void loginLengthLessThen_6_CharactersIsNotOk() {
+        User actual = user;
+        actual.setLogin("test");
+        assertThrows(UserNotAllowed.class, () -> registrationService.register(actual));
+    }
+
+    @Test
+    void ageLessThan_18_NotOk() {
+        User actual = user;
+        actual.setAge(12);
+        assertThrows(UserNotAllowed.class, () -> registrationService.register(actual));
+    }
+
+    @Test
+    void passwordLengthLessThan_6_characters_NotOk() {
+        User actual = user;
+        actual.setPassword("test");
+        assertThrows(UserNotAllowed.class, () -> registrationService.register(actual));
```

Comment:
The test method name `passwordLengthLessThan_6_characters_NotOk` does not follow the naming convention `<methodUnderTest>_<state>_<expectedBehavior>`. Consider renaming it to `register_passwordLengthLessThan6Characters_notOk`. Also, ensure that you're testing the edge cases for the password length as mentioned in the checklist.